packages/effect/src/Effect.ts

```diff
@@ -253,6 +253,11 @@ export declare namespace Effect {
    * @category Effect Type Extractors
    */
   export type Success<T extends Effect<any, any, any>> = [T] extends [Effect<infer _A, infer _E, infer _R>] ? _A : never
+  /**
+   * @since 3.15.5
+   * @category Effect Type Extractors
+   */
+  export type AsEffect<T extends Effect<any, any, any>> = {} & Effect<Success<T>, Error<T>, Context<T>>
```

Comment:
```suggestion
  export type AsEffect<T extends Effect<any, any, any>> = [T] extends [Effect<infer _A, infer _E, infer _R>] ? Effect<_A, _E, _R> : never
```

Does this work? Will be more direct this way.